src/avram/associations.cr

```diff
@@ -68,18 +67,26 @@ module Avram::Associations
       {% nilable = false %}
     {% end %}
 
-    column {{ assoc_name.id }}_id : {{ model }}::PrimaryKeyType{% if nilable %}?{% end %}
+    {% if !foreign_key %}
+      {% foreign_key = "#{assoc_name}_id".id %}
+    {% end %}
+
+    {% if !parent_table %}
+      {% parent_table = run("../run_macros/infer_table_name.cr", model.id) %}
```

Comment:
This will be a good spot for that `table_for`. Should that go in here? Or wait until we add it in to that model stuff later?

Reply:
What's `table_for`?

Reply:
Eating mostly

Reply:
But really, `table_for` doesn't work because we don't have access to it inside the macro interpolation. That's why I think it wasn't working on the other PR and I don't think there is a way around it.

@edwardloveall `table_for` was just added in #123 for migrations. It infers the table name based on the model class. It uses the same `run_macro` that is used ehre

Reply:
Actually. Maybe I can get it working. Let me try one more time :)

Reply:
Nope. Here is the issue. At macro compile time we just get the raw `table_for(MyModel)` not the result of that macro call. So it doesn't work for stuff like this.

Here's an example: https://play.crystal-lang.org/#/r/76bh

Reply:
ZING!